ganttproject/src/net/sourceforge/ganttproject/GPTreeTableBase.java

```diff
@@ -528,10 +536,80 @@ protected void initTreeTable() {
     doInit();
   }
 
+  private static Comparator<Task> beginComparatorAsc = new Comparator<Task>() {
```

Comment:
Please move comparators into the table column class. It was mentioned in the [design document](https://github.com/bardsoftware/ganttproject/wiki/Feature-Notes:-Task-Sorting):
> add sortComparator property to TaskDefaultColumn interface and its implementation in BEGIN_DATE and END_DATE

and it will encapsulate column logic in a single place.